src/System.ComponentModel.TypeConverter/src/System/ComponentModel/ReflectTypeDescriptionProvider.cs

```diff
@@ -90,53 +91,59 @@ internal ReflectTypeDescriptionProvider()
         {
         }
 
-        private static Hashtable InitIntrinsicTypeConverters()
-        {
-            // Add the intrinsics
-            return new Hashtable
-            {
-                [typeof(bool)] = typeof(BooleanConverter),
-                [typeof(byte)] = typeof(ByteConverter),
-                [typeof(SByte)] = typeof(SByteConverter),
-                [typeof(char)] = typeof(CharConverter),
-                [typeof(double)] = typeof(DoubleConverter),
-                [typeof(string)] = typeof(StringConverter),
-                [typeof(int)] = typeof(Int32Converter),
-                [typeof(short)] = typeof(Int16Converter),
-                [typeof(long)] = typeof(Int64Converter),
-                [typeof(float)] = typeof(SingleConverter),
-                [typeof(UInt16)] = typeof(UInt16Converter),
-                [typeof(UInt32)] = typeof(UInt32Converter),
-                [typeof(UInt64)] = typeof(UInt64Converter),
-                [typeof(object)] = typeof(TypeConverter),
-                [typeof(void)] = typeof(TypeConverter),
-                [typeof(DateTime)] = typeof(DateTimeConverter),
-                [typeof(DateTimeOffset)] = typeof(DateTimeOffsetConverter),
-                [typeof(Decimal)] = typeof(DecimalConverter),
-                [typeof(TimeSpan)] = typeof(TimeSpanConverter),
-                [typeof(Guid)] = typeof(GuidConverter),
-                [typeof(Uri)] = typeof(UriTypeConverter),
-                [typeof(Color)] = typeof(ColorConverter),
-                [typeof(Point)] = typeof(PointConverter),
-                [typeof(Rectangle)] = typeof(RectangleConverter),
-                [typeof(Size)] = typeof(SizeConverter),
-                [typeof(SizeF)] = typeof(SizeFConverter),
-                // Special cases for things that are not bound to a specific type
-                //
-                [typeof(Array)] = typeof(ArrayConverter),
-                [typeof(ICollection)] = typeof(CollectionConverter),
-                [typeof(Enum)] = typeof(EnumConverter),
-                [s_intrinsicNullableKey] = typeof(NullableConverter),
-            };
-        }
+        private static Hashtable EditorTables => LazyInitializer.EnsureInitialized(ref s_editorTables, () => new Hashtable(4));
```

Comment:
@stephentoub Sorry for asking too many questions. I wanted to understand how EnsureInitialized works and found its implementation in the coreclr repo. For the above overload, the implementation is at this [link](https://github.com/dotnet/coreclr/blob/c35e8dbc37e5380f46553510d0368aad04a677d2/src/mscorlib/src/System/Threading/LazyInitializer.cs#L124).

It uses `Volatile.Read` to load address of target (`s_editorTables` in this case). I think the Volatile.Read would introduce memory barrier? The ask in current task is to eliminate memory barriers which are a consequence of volatile fields. But then `EnsureInitialized` appears to rely on similar strategy involving volatile reads... Which appears to be contradictory to what we are trying to achieve.

Whereas, the `Lazy<T>.Value` implementation ([link](https://github.com/dotnet/coreclr/blob/c35e8dbc37e5380f46553510d0368aad04a677d2/src/mscorlib/src/System/Lazy.cs#L319)) appears to directly return the value once initialised under a lock.

I might be missing something, but I thought I would check with you. Thanks.